### DISCUSSION TITLE
Dream3d exported file can NOT be read by EBSD_reconstruction module

### DISCUSSION THREAD

**Discussion by zengfy-hust** [idaholab/moose#17851] [Q&A Modules: Phase field] [answered]
Hello Moose workers:
         I use dream3d to covert EBSD file to INL file .then used it as an initia condition for EBSD_reconstruction module in phase_field module .however ,when I chanege the innput file .Error message was shown as followed 
*** ERROR ***
The following error occurred in the object "ebsd_reader", of type "EBSDReader".

EBSD Data ouside of the domain declared in the header ([-850:-730], [-0.5:119.5], [0:0]) dim=2

Stack frames: 16
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: callMooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, MooseApp*)
3: EBSDReader::readFile()
4: EBSDReader::EBSDReader(InputParameters const&)
5: std::shared_ptr<MooseObject> moose::internal::buildObj<EBSDReader>(InputParameters const&)
6: Factory::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters&, unsigned int, bool)
7: FEProblemBase::addUserObject(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters&)
8: Action::timedAct()
9: ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
10: ActionWarehouse::executeAllActions()
11: MooseApp::runInputFile()
12: MooseApp::run()
13: /home/zengfanyi/projects/moose/modules/phase_field/phase_field-opt(+0x527e) [0x55d01cade27e]
14: __libc_start_main
15: /home/zengfanyi/projects/moose/modules/phase_field/phase_field-opt(+0x550d) [0x55d01cade50d]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
[EBSD_reconstruction.zip](https://github.com/idaholab/moose/files/6487487/EBSD_reconstruction.zip)

**Comment by dschwen** [comment]
The error message indicates that one of the data points has coordinates outside the slab defined in the file's header. I'll take a look at the file. Maybe we can just discard such points and not fuss about it. I'm surprised this has not come up before, though.

**Comment by dschwen** [comment]
Looks to me like the header is not well formed. For one, the `x_dim`, `y_dim`, `z_dim` values should all be _integers_. Also `x_dim * x_step + x_min` should give you `x_max`. Please double check your export options or try changing the file manually. E.g. `x_dim should be (1.5+850.5)/1.5 = 568.0`.

**Comment by zengfy-hust** [comment]
thank you ,I will follow your instruction this evening ,and I will let you know if it works well .

**Comment by zengfy-hust** [comment]
Hello Moose workers 👍 
         A new error message occured when I change the input file and the EBSD_reconstruction file .I attached both files in this reply just for your better understand for my pronblem.The error meaasge was shown as following :
*** ERROR ***
A 'EBDSMeshGenerator' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

Stack frames: 21
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)
3: Factory::reportUnregisteredError(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
4: Factory::getValidParams(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
5: MooseObjectAction::MooseObjectAction(InputParameters)
6: AddMeshGeneratorAction::AddMeshGeneratorAction(InputParameters)
7: /home/zengfanyi/projects/moose/framework/libmoose-opt.so.0(+0xf3a968) [0x7f80565dc968]
8: std::shared_ptr<Action> moose::internal::buildAct<AddMeshGeneratorAction>(InputParameters const&)
9: ActionFactory::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters&)
10: Parser::walkRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, hit::Node*)
11: Parser::walk(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, hit::Node*)
12: hit::Node::walk(hit::Walker*, hit::NodeType)
13: hit::Node::walk(hit::Walker*, hit::NodeType)
14: hit::Node::walk(hit::Walker*, hit::NodeType)
15: Parser::parse(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
16: MooseApp::setupOptions()
17: MooseApp::run()
18: /home/zengfanyi/projects/moose/modules/phase_field/phase_field-opt(+0x527e) [0x564a0ef9527e]
19: __libc_start_main
20: /home/zengfanyi/projects/moose/modules/phase_field/phase_field-opt(+0x550d) [0x564a0ef9550d]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
[Uploading EBSD_reconstruction.zip…]()

**Reply by GiudGiud** [reply]
Did you include phase field in the list of modules to compile for your application?

**Reply by zengfy-hust** [reply]
sure I did

**Comment by dschwen** [comment]
This is not related to the EBSD data file. You are running the input with an executable that does not have support for EBSD built in. I'm turning this into a discussion now. As this does not seem to be a MOOSE issue.

**Comment by GiudGiud** [comment, marked as answer]
Check the spelling. I think it's EBSD  MeshGenerator not EBDS

**Reply by zengfy-hust** [reply]
sorry  for the fault ,after I correct that ,another error occurred .
*** ERROR ***
The following error occurred in the object "ebsd_mesh", of type "EBSDMeshGenerator".

EBSDMeshGenerator error. Requested uniform_refine levels not possible.
I wonder what uniform_refine level should it be since I already set it to 1 .for your convenience ,I just upload the file I used for your better understanding of my question .FYI ,the coordination in EBSD_reconstruction.txt had been moved 0.5 just in case the coordination  overlapped. 
[EBSD_reconstruction.zip](https://github.com/idaholab/moose/files/6573091/EBSD_reconstruction.zip)
hope to hear from you soon .

**Reply by GiudGiud** [reply]
Please check the documentation for more explanation on this object and the uniform refine parameter
https://mooseframework.inl.gov/source/meshgenerators/EBSDMeshGenerator.html

**Reply by zengfy-hust** [reply]
Another error occurred ,Do you know why this happen ?
*** ERROR ***
The following error occurred in the object "ebsd_reader", of type "EBSDReader".

An EBSD feature needs to have a uniform phase.
